core/store/src/main/java/uk/gov/gchq/gaffer/store/StoreProperties.java

```diff
@@ -178,6 +178,7 @@ public static <T extends StoreProperties> T loadStoreProperties(final InputStrea
         return (T) updateInstanceType(requiredClass, properties);
     }
 
+    @SuppressWarnings("PMD.UseTryWithResources") //Not possible
```

Comment:
I think I'm missing something here - why is this not possible?

Reply:
The reason in similar to what I put in a response to an earlier comment. In this case it's because the `storePropertiesStream` already exists in the method and for Java 8 a try with resources required the resource to be declared in the try statement itself. If it has been declared already then try with resources cannot be used. This is improved in Java 9+, so if were using that then we could use try with resources here.